packages/stargate/src/signingstargateclient.ts

```diff
@@ -120,8 +119,26 @@ export class SigningStargateClient extends StargateClient {
       value: txBody,
     });
     const gasLimit = Int53.fromString(fee.gas).toNumber();
-    const authInfoBytes = makeAuthInfoBytes([pubkeyAny], fee.amount, gasLimit, sequence);
 
+    if (!isOfflineDirectSigner(this.signer)) {
+      const signMode = cosmos.tx.signing.v1beta1.SignMode.SIGN_MODE_LEGACY_AMINO_JSON;
+      const authInfoBytes = makeAuthInfoBytes([pubkeyAny], fee.amount, gasLimit, sequence, signMode);
+      const msgs = messages.map((msg) => ({
+        type: "cosmos-sdk/MsgDelegate",
```

Comment:
I guess we need a whole lookup table of amino names to proto names? I searched through and there doesn't seem to be an explicit map. But you can look in the various `codec.go` files and see the amino names attached to a struct. The protoname should be the same as the struct. I guess a mapping of proto.Any name -> amino name would be sufficient for this to work?

https://github.com/cosmos/cosmos-sdk/blob/release/v0.40.x/x/staking/types/codec.go#L12-L18

Reply:
But other than that (WIP) point, this looks like a good approach